dspace-api/src/main/java/org/dspace/xmlworkflow/service/XmlWorkflowService.java

```diff
@@ -41,39 +41,171 @@
+     * @param workflowItemId the workflow item on which to take the action.
+     * @param workflow the workflow holding the item.
+     * @param currentActionConfig the requested action.
+     * @return the next action to be executed.
+     * @throws SQLException passed through.
+     * @throws AuthorizeException if the user may not take this action.
+     * @throws IOException passed through.
+     * @throws MessagingException unused.
+     * @throws WorkflowException if the action could not be executed.
+     */
     public WorkflowActionConfig doState(Context c, EPerson user, HttpServletRequest request, int workflowItemId,
                                         Workflow workflow, WorkflowActionConfig currentActionConfig)
         throws SQLException, AuthorizeException, IOException, MessagingException, WorkflowException;
 
+    /**
+     * Select the next action based on the outcome of a current action.
+     *
+     * @param c session context.
+     * @param user current user.
+     * @param workflow item is in this workflow.
+     * @param currentStep workflow step being executed.
+     * @param currentActionConfig describes the current step's action.
+     * @param currentOutcome the result of executing the current step (accept/reject/etc).
+     * @param wfi the Item being processed through workflow.
+     * @param enteredNewStep is the Item advancing to a new workflow step?
+     * @return the next step's action.
+     * @throws SQLException passed through.
+     * @throws AuthorizeException passed through.
+     * @throws IOException passed through.
+     * @throws WorkflowException if the current step's outcome is unrecognized.
+     */
     public WorkflowActionConfig processOutcome(Context c, EPerson user, Workflow workflow, Step currentStep,
                                                WorkflowActionConfig currentActionConfig, ActionResult currentOutcome,
                                                XmlWorkflowItem wfi, boolean enteredNewStep)
         throws IOException, AuthorizeException, SQLException, WorkflowException;
 
+    /**
+     * Deletes all tasks from a WorkflowItem.
+     *
+     * @param context current DSpace session
+     * @param wi      the workflow item for which we are to delete the tasks
+     * @throws SQLException       passed through.
+     * @throws AuthorizeException passed through.
+     */
     public void deleteAllTasks(Context context, XmlWorkflowItem wi) throws SQLException, AuthorizeException;
 
+    /**
+     * Deletes all pooled tasks from a WorkflowItem.
+     * @param c  current DSpace session.
+     * @param wi the workflow item from which we are to delete the tasks.
+     * @throws SQLException       passed through.
+     * @throws AuthorizeException passed through.
+     */
     public void deleteAllPooledTasks(Context c, XmlWorkflowItem wi) throws SQLException, AuthorizeException;
 
+    /**
+     * Deletes an EPerson from the task pool of a step.
```

Comment:
probably the confusion is due to the fact that tasks are usually in 1:1 relation with eperson. That said I agree with @tdonohue it would be better to rephrase

Reply:
Well caught.  I don't know what I was thinking when I wrote that this was about `EPerson`s.  Probably I had been staring for too long at too much undocumented code.